File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
#규칙에 맞지 않는 아이디를 입력했을 때, 입력한 아이디와 유사하면서 규칙에 맞는 아이디 추천
import re
def solution(new_id):
    str=new_id.lower()
    str=re.sub('[^a-z0-9\-_.]', '',str)
    str=re.sub('\.+', '.', str)
    str=re.sub('^[.]|[.]$', '',str)
    if str=='':
        str='a'
    str=str[:15]
    str=re.sub('[.]$','',str)
    while len(str) < 3:
        str += str[-1]
    return str

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    let id = new_id;
    
    // 1단계
    id = id.toLowerCase();
    
    // 2단계 - null 문제 해결
    id = id.replace(/[^a-z0-9-_.]/g, '');
    
    // 3단계
    id = id.replace(/\.+/g, '.');
    
    // 4단계
    id = id.replace(/^\.|\.$/g, '');
    
    // 5단계
    if (id.length === 0) {
        id = 'a';
    }
    
    // 6단계
    if (id.length >= 16) {
        id = id.slice(0, 15).replace(/\.$/, '');
    }
    
    // 7단계 - 현재 id를 기준으로 체크
    while (id.length <= 2) {
        id += id[id.length - 1];
    }
    
    return id;
}